### ISSUE TITLE
can you reference a graphviz with numref?

### ISSUE THREAD

**Issue by punsoong** [sphinx-doc/sphinx#3136]
Is it possible to reference a graphviz directive using numref or even ref?
e.g.
```
The giagram .. :numref:`a_diagram`

.. graphviz::
   :caption: this is the caption
   :name: a_diagram
   
   digraph foo {
      "bar" -> "baz";
   }
```

This fails since the graphviz doesn't support the :name: option.

Thanks

**Comment by tk0miya** [comment]
Indeed, `graphviz` directive does not support `:name:` option yet.
But you can do it with following notation:

```
The giagram .. :numref:`a_diagram`

.. _a_diagram:
.. graphviz::
   :caption: this is the caption

   digraph foo {
      "bar" -> "baz";
   }
```

**Comment by tk0miya** [comment]
Now I added `:name:` option to graphviz directives.
It will be released as Sphinx-1.6.

Thanks!
